src/Config.AzureKeyVault/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -38,7 +38,7 @@ public AzureKeyVaultConfigurationProvider(IKeyVaultClient client, string vault,
             _manager = manager;
         }
 
-        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+        public override void Load() => Task.Run(async() => await LoadAsync()).Wait();
```

Comment:
I don't believe this is the correct fix, the correct fix is to add .ConfigureAwait(false) to the additional awaits (lines 47, 57, 63 etc). I tested that in my failing example and it fixes the issue

Reply:
It seems to be a viable solution for my failing example as well, I have updated the pull request with your suggested changes!

Thanks a lot for taking time to review this pull request! :)